lib/install.js

```diff
@@ -382,11 +383,13 @@ Installer.prototype.loadAllDepsIntoIdealTree = function (cb) {
     steps.push([validateArgs, this.idealTree, this.args])
     steps.push([loadRequestedDeps, this.args, this.idealTree, saveDeps, cg.newGroup('loadRequestedDeps')])
   } else {
-    if (this.prod) {
+    if (this.prod && this.dev) {
```

Comment:
I assume you're viewing this as related to your second condition, but I'm not entirely comfortable with this approach. Splitting the prod and dev steps is intentional and the idea is that:
```
npm install --only=prod
npm install --only=dev
```
should give you exactly the same tree as:
```
npm install
```
And I think your change is intentionally breaking that?